Update mockups to support Space Engineers v1.190

kwilliams1987 opened the pull request:
- Added mockup for `IMyTextSurface`.
  - Required for mocking `IMyProgrammableBlock` and `IMyTextPanel`.
- Added several new mocked blocks.
  - `IMyRotorRotor` (Rotor Head)
  - `IMyRotorStator` (Rotor Body)
  - `IMyDoor` (Sliding Door)
- Started implementing mock inventory systems.
- Improved the `Assert` utility to closer match standard unit test Asserts.

malware-dev commented on Mockups/Base/MockCubeBlock.debug.cs:
```diff
@@ -71,12 +62,12 @@ public override string ToString()
+            if (!string.IsNullOrWhiteSpace(DisplayName))
             {
-                name += ": " + DisplayNameText;
+                name += ": " + DisplayName;
```
Why was this done?

kwilliams1987 replied:
`DisplayNameText` wasn't set during mock runs, whereas `DisplayName` was.

malware-dev commented on Mockups/Asserts/Assert.debug.cs:
```diff
@@ -1,11 +1,50 @@
+            throw new AssertionException(message);
+        }
+
+        public static void True(bool condition, string message)
```
These changes goes against my original plan for this file, which was to create an NUnit/FluentAssertions type syntax, but that's OK.

kwilliams1987 replied:
I based these off the standard MSTest Asserts, but if you prefer a different language I can change them.

malware-dev commented on Mockups/Base/MockCubeBlock.debug.cs:
```diff
@@ -43,19 +42,11 @@ public abstract class MockCubeBlock : MockEntity, IMyCubeBlock
-        {
-            throw new NotImplementedException();
-        }
+        public virtual void UpdateVisual() { }
```
These methods shouldn't be used by a script.

kwilliams1987 replied:
Would make more sense to throw `NotSupportedException` or `InvalidOperationException` I guess?
Optionally could flag them as `[Obsolete("Do not call from ingame scripts.", true)]` to prevent them even being referenced?

malware-dev commented on Mockups/Base/MockTerminalBlock.debug.cs:
```diff
@@ -97,13 +100,29 @@ public virtual bool HasLocalPlayerAccess()
         public virtual bool HasPlayerAccess(long playerId)
         {
-            throw new NotImplementedException();
+            switch (Relationships.GetValueOrDefault(playerId, MyRelationsBetweenPlayerAndBlock.Neutral))
```
This method has no meaning for scripts.

kwilliams1987 replied:
I will revert and comment and tag as not usable in scripts.

malware-dev commented on Mockups/MockedRun.debug.cs:
```diff
@@ -23,7 +23,7 @@ protected MockedRun()
         /// Determines whether this run has been initialized. 
         /// </summary>
-        public bool IsInitialized { get; private set; }
+        public bool IsInitialized { get; set; }
```
It's been a while, but I have a feeling this is wrong. This should probably not be settable from the outside.
- Yep, confirmed. This is not a good change.

kwilliams1987 commented:
I've added `[Obsolete("This method should not be referenced by ingame scripts.", true)]` attributes to methods which shouldn't be accessed by ingame scripts.

It feels a bit icky, but a compile time error that you're trying to perform an illegal action is better than thinking it should work and getting runtime errors in game.

I've also added the new DataExtractor tool. This tool allows accurate extraction of ingame variables (such as font and sprite lists) and addes them to generated partial classes.

I'll expand on this tool for other classes containing data lists from in-game content later.
